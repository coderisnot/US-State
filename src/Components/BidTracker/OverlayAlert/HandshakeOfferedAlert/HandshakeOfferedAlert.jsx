import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

class HandshakeOfferedAlert extends Component {
  constructor(props) {
    super(props);
    this.onAcceptBid = this.onAcceptBid.bind(this);
    this.onDeclineBid = this.onDeclineBid.bind(this);
  }
  onAcceptBid() {
    const { acceptBid, id } = this.props;
    acceptBid(id);
  }
  onDeclineBid() {
    const { declineBid, id } = this.props;
    declineBid(id);
  }
  render() {
    const { userName } = this.props;
    return (
      <div className="bid-tracker-alert-container bid-tracker-alert-container--handshake-offered">
        <div className="top-text">{`${userName}, you've been offered a handshake`}</div>
        <div className="usa-grid-full">
          <button className="tm-button-transparent" onClick={this.onAcceptBid}>
            <FontAwesome name="check-o" /> Accept Handshake
          </button>
          <button className="tm-button-transparent tm-button-no-box" onClick={this.onDeclineBid}>
            Decline Handshake
          </button>
        </div>
        <div className="sub-text">24 hours to accept the handshake</div>
      </div>
    );
  }
}

HandshakeOfferedAlert.propTypes = {
  id: PropTypes.number.isRequired,
  userName: PropTypes.string.isRequired,
  acceptBid: PropTypes.func.isRequired,
  declineBid: PropTypes.func.isRequired,
};

export default HandshakeOfferedAlert;
