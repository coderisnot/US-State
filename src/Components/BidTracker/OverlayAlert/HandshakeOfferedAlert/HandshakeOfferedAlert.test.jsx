import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import HandshakeOfferedAlert from './HandshakeOfferedAlert';

describe('HandshakeOfferedAlertComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <HandshakeOfferedAlert id={1} userName="test" acceptBid={() => {}} declineBid={() => {}} />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can accept a bid', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <HandshakeOfferedAlert id={1} userName="test" acceptBid={spy} declineBid={() => {}} />,
    );
    const button = wrapper.find('button').at(0);
    button.simulate('click');
    sinon.assert.calledOnce(spy);
  });

  it('can decline a bid', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <HandshakeOfferedAlert id={1} userName="test" acceptBid={() => {}} declineBid={spy} />,
    );
    const button = wrapper.find('button').at(1);
    button.simulate('click');
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <HandshakeOfferedAlert id={1} userName="test" acceptBid={() => {}} declineBid={() => {}} />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
