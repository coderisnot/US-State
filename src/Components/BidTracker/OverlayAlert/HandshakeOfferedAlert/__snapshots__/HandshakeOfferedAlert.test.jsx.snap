// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandshakeOfferedAlertComponent matches snapshot 1`] = `
<div
  className="bid-tracker-alert-container bid-tracker-alert-container--handshake-offered"
>
  <div
    className="top-text"
  >
    test, you've been offered a handshake
  </div>
  <div
    className="usa-grid-full"
  >
    <button
      className="tm-button-transparent"
      onClick={[Function]}
    >
      <FontAwesome
        name="check-o"
      />
       Accept Handshake
    </button>
    <button
      className="tm-button-transparent tm-button-no-box"
      onClick={[Function]}
    >
      Decline Handshake
    </button>
  </div>
  <div
    className="sub-text"
  >
    24 hours to accept the handshake
  </div>
</div>
`;
